src/gmp/z.d

```diff
@@ -588,6 +589,7 @@ nothrow:
             else static if (s == "|") __gmpz_ior(y._ptr, _ptr, rhs._ptr);
             else static if (s == "^") __gmpz_xor(y._ptr, _ptr, rhs._ptr);
             else static if (s == "<<") __gmpz_mul_2exp(y._ptr, _ptr, cast(ulong)rhs);
+            else static if (s == ">>") __gmpz_tdiv_q_2exp(y._ptr, _ptr, cast(ulong)rhs);
```

Comment:
We need to handle the cases where `rhs` is negative for both `<<` and `>>`, right?

Reply:
Mmmh right.

If `rhs` is negative, shall I disallow it (`assert(0);`), or simply reconverting it to the opposite sign (`z >> -1` becomes `z << 1` ?

Reply:
I vote for the latter.